src/Extractor/Snowflake.php

```diff
@@ -57,88 +37,32 @@ protected function createExportAdapter(): ExportAdapter
             $this->logger,
             $this->connection,
             $this->getQueryFactory(),
-            $this->getDatabaseConfig(),
-            $this->snowSqlConfig,
-            $this->temp,
-            $this->getMetadataProvider()
+            $this->getDatabaseConfig()
         );
     }
 
     protected function getQueryFactory(): SnowflakeQueryFactory
     {
         if (!isset($this->queryFactory)) {
-            $this->queryFactory = new SnowflakeQueryFactory($this->state);
+            $this->queryFactory = new SnowflakeQueryFactory(
+                $this->connection,
+                $this->getMetadataProvider(),
+                $this->state
+            );
         }
 
         return $this->queryFactory;
     }
 
     public function createConnection(DatabaseConfig $databaseConfig): void
     {
-        if (!($databaseConfig instanceof SnowflakeDatabaseConfig)) {
-            throw new ApplicationException('Instance of SnowflakeDatabaseConfig exceded');
-        }
-        $this->snowSqlConfig = $this->createSnowSqlConfig($databaseConfig);
-
-        if ($databaseConfig->hasWarehouse()) {
-            $this->warehouse = $databaseConfig->getWarehouse();
-        }
-
-        try {
-            $this->connection = new SnowflakeOdbcConnection(
-                $this->logger,
-                $databaseConfig
-            );
-            if ($databaseConfig->hasSchema()) {
-                $this->connection->query(
-                    sprintf(
-                        'USE SCHEMA %s.%s',
-                        $this->connection->quoteIdentifier($databaseConfig->getDatabase()),
-                        $this->connection->quoteIdentifier($databaseConfig->getSchema())
-                    )
-                )->closeCursor();
-                $this->logger->info(sprintf(
-                    'Use schema "%s" in database "%s"',
-                    $databaseConfig->getSchema(),
-                    $databaseConfig->getDatabase()
-                ));
-            }
-            if (getenv('KBC_RUNID')) {
-                $queryTag = ['runId' => getenv('KBC_RUNID')];
-
-                $this->connection->query("ALTER SESSION SET QUERY_TAG='" . json_encode($queryTag) . "';");
-            }
-        } catch (CannotAccessObjectException $e) {
-            throw new UserException($e->getMessage(), 0, $e);
-        }
+        $factory = new SnowflakeConnectionFactory($this->logger);
```

Comment:
Bol tu historicky dost chaos ako sa vytvara a setupuje spojenie, presunul som to do factory.

Hlavny problem bol, ze sa spojenie setupovalo az po jeho vytvoreni, nepouzival sa `init` callback, ... to znamena, ze ak by spojenie spadlo a nastalo by retry, ... tak by dalsie spojenie uz nebolo spravne nastavene, ... napr. by nemalo nastaveny `queryTag`.